dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -1353,6 +1385,23 @@ CFileDefExt::CountFolderAndFiles(HWND hwndDlg, LPWSTR pwszBuf, UINT cchBufMax, D
             FileSize.u.LowPart  = wfd.nFileSizeLow;
             FileSize.u.HighPart = wfd.nFileSizeHigh;
             m_DirSize.QuadPart += FileSize.QuadPart;
+            // Calculate size on disc
+            // if disk usage is not aligned on a complete disk geometry block (sector/cluster), align on the next block size, depending on disk geometry data
+            // This could be improved as per : https://docs.microsoft.com/en-us/previous-versions/technet-magazine/hh148159(v=msdn.10)
+            if (GetVolumePathName(pwszBuf, szVolumePathName, _countof(szVolumePathName)))
+            {
+                if (GetDiskFreeSpace(szVolumePathName, &ulSectorsPerCluster, &ulBytesPerSector, NULL, NULL))
+                {
+                    ulBlockSize = ulBytesPerSector*ulSectorsPerCluster;
+                    m_DirSizeOnDisc.QuadPart += FileSize.QuadPart % (ulBlockSize) ? ((FileSize.QuadPart/(ulBlockSize))+1)*ulBlockSize : FileSize.QuadPart;
```

Comment:
```suggestion
                    m_DirSizeOnDisc.QuadPart += FileSize.QuadPart;
                    if (FileSize.QuadPart % ulBlockSize)
                        m_DirSizeOnDisc.QuadPart += ulBlockSize - FileSize.QuadPart % ulBlockSize;
```

Reply:
For the sake of lisibility of the formula, I prefer the original, while I reformat it in a more compliant way

Reply:
Your calculation is wrong when the file size was `ulBlockSize`

Reply:
Will be refactored using NtQueryInformationFile with FileStandardInformation